src/ProtocolCollection.php

```diff
@@ -96,7 +96,6 @@ public static function throwIfUnsupported(ProtocolInterface $protocol)
      */
     public static function protocolFromHeaderPart(string $headerPart): ProtocolInterface {
         if (empty(self::$headerLookup)) {
-            /** @var ProtocolInterface $protocolClass */
```

Comment:
This messes IDEs, but Psalm correctly infer type from `@const array<int, class-string<ProtocolInterface>>` notation added above